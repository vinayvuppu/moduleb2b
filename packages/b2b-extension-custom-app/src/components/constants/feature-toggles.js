import SHARED_FLAGS from '@commercetools-local/constants/feature-toggles';

export const STORES_IN_ORDER_CREATE = 'storesInOrderCreate';

/*
 * This file contains a list of all feature toggles used within
 * this plugin. Each feature toggle should be a constant `.
 *
 * NOTE:
 *   Default values are not yet passed to flopflip and it has to be decided
 *   how the integrate with the plugin API.
 *   The default values would be used whenever feature toggles have not been
 *   fetched from LaunchDarkly (without default values flopflip will default
 *   them to off until fetched).
 */

// eslint-disable-next-line import/prefer-default-export
export const FLAGS = {
  ...SHARED_FLAGS,
  STORES_IN_ORDER_CREATE,
};
