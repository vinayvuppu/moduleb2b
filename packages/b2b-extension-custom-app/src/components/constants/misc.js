export const SEARCH_SLICE_NAME_ORDERS = 'application-orders/orders';
export const SEARCH_SLICE_NAME_ORDER_CREATE_CUSTOMERS =
  'application-orders/orders/new/customer';

// The Payment API has to accept uppercased and
// capitalized variations of transation state and types.
// This maps between them to render the messages accordingly.
export const TRANSACTION_TYPES = {
  AUTHORIZATION: 'Authorization',
  CANCELAUTHORIZATION: 'CancelAuthorization',
  CHARGE: 'Charge',
  REFUND: 'Refund',
  CHARGEBACK: 'Chargeback',
};

export const TRANSACTION_STATES = {
  PENDING: 'Pending',
  SUCCESS: 'Success',
  FAILURE: 'Failure',
  INITIAL: 'Initial',
};

// Shipment states for the return management tab
export const RETURN_SHIPMENT_INITIAL_STATES = {
  ADVISED: 'Advised',
  RETURNED: 'Returned',
};

export const RETURN_SHIPMENT_STATES = {
  ADVISED: 'Advised',
  RETURNED: 'Returned',
  BACK_IN_STOCK: 'BackInStock',
  UNUSABLE: 'Unusable',
};

export const RETURN_SHIPMENT_STATES_TRANSITIONS = {
  RETURNED: 'Returned',
  BACK_IN_STOCK: 'BackInStock',
  UNUSABLE: 'Unusable',
};

// Payment states for the return management tab
export const RETURN_PAYMENT_STATES = {
  INITIAL: 'Initial',
  REFUNDED: 'Refunded',
  NOT_REFUNDED: 'NotRefunded',
  NON_REFUNDABLE: 'NonRefundable',
};

export const RETURN_PAYMENT_STATES_TRANSITIONS = {
  INITIAL: 'Initial',
  REFUNDED: 'Refunded',
  NOT_REFUNDED: 'NotRefunded',
};

export const ORDER_STATES_VISIBILITY = {
  HIDE_ORDER_STATE: 'HideOrderState',
  HIDE_PAYMENT_STATE: 'HidePaymentState',
  HIDE_SHIPMENT_STATE: 'HideShipmentState',
};

// eslint-disable-next-line import/prefer-default-export
export const FIELD_TYPES = {
  Money: 'Money',
  LocalizedString: 'LocalizedString',
  Time: 'Time',
  DateTime: 'DateTime',
  Boolean: 'Boolean',
  String: 'String',
  Enum: 'Enum',
  LocalizedEnum: 'LocalizedEnum',
  Number: 'Number',
  Date: 'Date',
  Reference: 'Reference',
  Set: 'Set',
};

export const DISCOUNT_VALUE_TYPES = {
  ABSOLUTE: 'absolute',
  RELATIVE: 'relative',
};
