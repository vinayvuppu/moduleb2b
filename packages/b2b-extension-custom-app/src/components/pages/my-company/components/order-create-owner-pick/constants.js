// eslint-disable-next-line
export const OWNER_TYPES = {
  OWNER_ME: 'me',
  OWNER_EMPLOYEE: 'employee',
  OWNER_COMPANY: 'company',
};
