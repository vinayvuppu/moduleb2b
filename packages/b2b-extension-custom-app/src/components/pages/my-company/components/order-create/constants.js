// Tab names for the steps in the header which should be defined
// in order because they determine the animation direction when
// switching between tabs
// eslint-disable-next-line import/prefer-default-export
export const ORDER_CREATE_TAB_NAMES = {
  ADD_LINE_ITEMS: 'lineitems',
  SET_OWNER: 'owner',
  SET_SHIPPING_METHOD: 'shipping-method',
  CONFIRMATION: 'confirmation',
};

export const ORDER_STATES = {
  OPEN: 'open',
  PENDING_APPROVAL: 'pendingApproval',
};
