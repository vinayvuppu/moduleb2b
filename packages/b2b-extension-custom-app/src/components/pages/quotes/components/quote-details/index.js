export { default } from './quote-details';

export * from './quote-details';
