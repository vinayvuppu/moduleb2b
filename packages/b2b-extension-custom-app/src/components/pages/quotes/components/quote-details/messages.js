import { defineMessages } from 'react-intl';

export default defineMessages({
  backToList: {
    id: 'Quote.Details.backToList',
    defaultMessage: 'To quotes list',
  },
  title: {
    id: 'Quote.Details.title',
    defaultMessage: 'Quote {quoteNumber}',
  },
  approvalButton: {
    id: 'Quote.Details.approvalButton',
    defaultMessage: 'Approve',
  },
  rejectButton: {
    id: 'Quote.Details.rejectButton',
    defaultMessage: 'Decline',
  },
  tabGeneral: {
    id: 'Quote.Update.tab.general',
    defaultMessage: 'General',
  },
  tabComments: {
    id: 'Quote.Update.tab.tabComments',
    defaultMessage: 'Comments',
  },
  confirmUpdateTitle: {
    id: 'Quote.Details.confirmUpdateTitle',
    defaultMessage: 'Update quote state',
  },
  confirmDeleteMessage: {
    id: 'Quote.Details.confirmDeleteMessage',
    defaultMessage: 'Are you sure you want to change the quote state?',
  },
  updateQuoteError: {
    id: 'Quote.Details.updateQuoteError',
    defaultMessage: 'An error occurred while updating the quote status',
  },
  approveSuccess: {
    id: 'Quote.Details.approveSuccess',
    defaultMessage: 'Quote approved',
  },
  rejectSuccess: {
    id: 'Quote.Details.rejectSuccess',
    defaultMessage: 'Quote declined',
  },
  placeOrder: {
    id: 'Quote.Details.placeOrder',
    defaultMessage: 'Place order',
  },
  editQuote: {
    id: 'Quote.Details.editQuote',
    defaultMessage: 'Edit',
  },
  submitQuote: {
    id: 'Quote.Details.submitQuote',
    defaultMessage: 'Submit',
  },
  submittedSuccess: {
    id: 'Quote.Details.submittedSuccess',
    defaultMessage: 'Quote submitted',
  },
  submittedError: {
    id: 'Quote.Details.submittedSuccess',
    defaultMessage: 'Quote submitted error',
  },
});
