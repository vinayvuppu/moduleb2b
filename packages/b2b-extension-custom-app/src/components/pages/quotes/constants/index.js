// eslint-disable-next-line
export const QUOTE_TYPES = {
  INITIAL: 'initial',
  SUBMITTED: 'submitted',
  APPROVED: 'approved',
  CLOSED: 'closed',
  DECLINED: 'declined',
  EXPIRED: 'expired',
  PLACED: 'placed',
};

export const QUOTE_DETAILS_TAB_NAMES = {
  GENERAL: 'general',
  COMMENTS: 'comments',
};
