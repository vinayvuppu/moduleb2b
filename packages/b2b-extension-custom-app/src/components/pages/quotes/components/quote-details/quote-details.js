import PropTypes from 'prop-types';
import React, { useState, Fragment, useContext } from 'react';
import { connect } from 'react-redux';
import oneLineTrim from 'common-tags/lib/oneLineTrim';
import { useIntl, FormattedMessage } from 'react-intl';
import { useRouteMatch, useHistory } from 'react-router-dom';

import {
  LoadingSpinner,
  Spacings,
  SecondaryButton,
  ReviewIcon,
  Text,
  PrimaryButton,
  CheckInactiveIcon,
  CartIcon,
} from '@commercetools-frontend/ui-kit';
import { RestrictedByPermissions } from '@commercetools-frontend/permissions';

import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
import BackToList from '@commercetools-local/core/components/back-to-list';
import * as globalActions from '@commercetools-frontend/actions-global';
import View from '@commercetools-local/core/components/view';
import ViewHeader from '@commercetools-local/core/components/view-header';
import TabHeader from '@commercetools-local/core/components/tab-header';
import TabContainer from '@commercetools-local/core/components/tab-container';
import {
  PageNotFound,
  ConfirmationDialog,
} from '@commercetools-frontend/application-components';
import { DOMAINS } from '@commercetools-frontend/constants';
import { ORDER_CREATE_TAB_NAMES } from '../../../my-company/components/order-create/constants';
import { PERMISSIONS } from '../../../../../constants';
import messages from './messages';
import { QUOTE_TYPES, QUOTE_DETAILS_TAB_NAMES } from '../../constants';

import EmployeeDetailsConnector from '../../../my-company/components/employee-detail-connector';
import B2BApolloClientContext from '../../../../common/b2b-apollo-client-context/context';

export const QuoteDetails = props => {
  const intl = useIntl();
  const routeMatch = useRouteMatch();
  const history = useHistory();
  const { formatMessage } = intl;

  const [confirmDialogOpen, setConfirmDialog] = useState(false);
  const [quoteStateAction, setQuoteStateAction] = useState();

  const closeDialog = () => {
    setConfirmDialog(false);
    setQuoteStateAction();
  };

  const openApproveDialog = () => {
    setQuoteStateAction(QUOTE_TYPES.APPROVED);
    setConfirmDialog(true);
  };

  const openRejectDialog = () => {
    setQuoteStateAction(QUOTE_TYPES.DECLINED);
    setConfirmDialog(true);
  };

  const handleGoPlaceOrder = () => {
    history.push(
      oneLineTrim(
        `/${routeMatch.params.projectKey}
        /b2b-extension
        /my-company
        /orders/new
        /${ORDER_CREATE_TAB_NAMES.ADD_LINE_ITEMS}`
      ),
      { quote: props.quote }
    );
  };

  const handleEditQuote = () => {
    history.push(
      oneLineTrim(
        `/${routeMatch.params.projectKey}
        /b2b-extension
        /my-company
        /quotes/new
        /${ORDER_CREATE_TAB_NAMES.ADD_LINE_ITEMS}`
      ),
      { quote: props.quote }
    );
  };

  const handleRequestQuote = async () => {
    try {
      await props.updateQuoteState(QUOTE_TYPES.SUBMITTED);
      props.showNotification({
        kind: 'success',
        domain: DOMAINS.SIDE,
        text: formatMessage(messages.submittedSuccess),
      });
    } catch (e) {
      props.showNotification({
        kind: 'error',
        domain: DOMAINS.SIDE,
        text: formatMessage(messages.submittedError),
      });
    }
  };

  const handleUpdateQuoteState = async () => {
    try {
      await props.updateQuoteState(quoteStateAction);
      props.showNotification({
        kind: 'success',
        domain: DOMAINS.SIDE,
        text: formatMessage(
          QUOTE_TYPES.APPROVED === quoteStateAction
            ? messages.approveSuccess
            : messages.rejectSuccess
        ),
      });
    } catch (error) {
      props.showNotification({
        kind: 'error',
        domain: DOMAINS.SIDE,
        text: formatMessage(messages.updateQuoteError),
      });
    }
    closeDialog();
  };

  if (props.isLoading) return <LoadingSpinner />;

  if (!props.quote) return <PageNotFound />;

  return (
    <div data-track-component="QuoteDetail">
      <View>
        <ViewHeader
          title={formatMessage(messages.title, {
            quoteNumber: props.quote.quoteNumber,
          })}
          backToList={
            <RestrictedByPermissions permissions={[PERMISSIONS.ViewOrders]}>
              <BackToList
                to={props.goToListRoute}
                label={formatMessage(messages.backToList)}
              />
            </RestrictedByPermissions>
          }
          commands={
            <Fragment>
              <Spacings.Inline
                scale="m"
                justifyContent="flex-end"
                alignItems="flex-end"
              >
                {QUOTE_TYPES.INITIAL === props.quote.quoteState &&
                  props.hasCompany && (
                    <RestrictedByPermissions
                      permissions={[PERMISSIONS.ManageOrders]}
                    >
                      <Fragment>
                        <SecondaryButton
                          label={formatMessage(messages.editQuote)}
                          onClick={handleEditQuote}
                        />
                        <PrimaryButton
                          label={formatMessage(messages.submitQuote)}
                          onClick={handleRequestQuote}
                        />
                      </Fragment>
                    </RestrictedByPermissions>
                  )}
                {QUOTE_TYPES.APPROVED === props.quote.quoteState &&
                  props.hasCompany && (
                    <RestrictedByPermissions
                      permissions={[PERMISSIONS.ManageOrders]}
                    >
                      <SecondaryButton
                        iconLeft={<CartIcon />}
                        label={formatMessage(messages.placeOrder)}
                        onClick={handleGoPlaceOrder}
                      />
                    </RestrictedByPermissions>
                  )}
                {QUOTE_TYPES.SUBMITTED === props.quote.quoteState && (
                  <RestrictedByPermissions
                    permissions={[
                      PERMISSIONS.ManageCompanies,
                      PERMISSIONS.ManageOrders,
                    ]}
                  >
                    <Fragment>
                      <SecondaryButton
                        iconLeft={<ReviewIcon />}
                        label={formatMessage(messages.approvalButton)}
                        onClick={openApproveDialog}
                      />
                      <PrimaryButton
                        iconLeft={<CheckInactiveIcon />}
                        tone="urgent"
                        label={formatMessage(messages.rejectButton)}
                        onClick={openRejectDialog}
                      />
                    </Fragment>
                  </RestrictedByPermissions>
                )}
              </Spacings.Inline>
            </Fragment>
          }
        >
          <TabHeader
            key={QUOTE_DETAILS_TAB_NAMES.GENERAL}
            name={QUOTE_DETAILS_TAB_NAMES.GENERAL}
            to={`${routeMatch.url}/${QUOTE_DETAILS_TAB_NAMES.GENERAL}`}
          >
            {formatMessage(messages.tabGeneral)}
          </TabHeader>
          <TabHeader
            key={QUOTE_DETAILS_TAB_NAMES.COMMENTS}
            name={QUOTE_DETAILS_TAB_NAMES.COMMENTS}
            to={`${routeMatch.url}/${QUOTE_DETAILS_TAB_NAMES.COMMENTS}`}
          >
            {formatMessage(messages.tabComments)}
          </TabHeader>
        </ViewHeader>
        <TabContainer>
          {props.children({
            hasCompany: props.hasCompany,
            employeeEmail: props.employeeEmail,
          })}
        </TabContainer>
      </View>
      <ConfirmationDialog
        title={formatMessage(messages.confirmUpdateTitle)}
        isOpen={confirmDialogOpen}
        labelPrimary={
          quoteStateAction === QUOTE_TYPES.APPROVED
            ? messages.approvalButton
            : messages.rejectButton
        }
        onClose={closeDialog}
        onCancel={closeDialog}
        onConfirm={handleUpdateQuoteState}
      >
        <Text.Body>
          <FormattedMessage {...messages.confirmDeleteMessage} />
        </Text.Body>
      </ConfirmationDialog>
    </div>
  );
};

const mapDispatchToProps = {
  showNotification: globalActions.showNotification,
};

const QuoteDetailsWrapper = props => {
  const {
    project: { key },
  } = useApplicationContext();
  const { apolloClient } = useContext(B2BApolloClientContext);
  return (
    <EmployeeDetailsConnector projectKey={key} apolloClient={apolloClient}>
      {({ employeeFetcher: { isLoading, employee } }) =>
        isLoading ? (
          <LoadingSpinner />
        ) : (
          <QuoteDetails
            {...props}
            hasCompany={Boolean(employee)}
            employeeEmail={employee?.email}
          />
        )
      }
    </EmployeeDetailsConnector>
  );
};
QuoteDetailsWrapper.displayName = 'QuoteDetailsWrapper';

QuoteDetails.displayName = 'QuoteDetails';
QuoteDetails.propTypes = {
  quoteId: PropTypes.string.isRequired,
  quote: PropTypes.object,
  isLoading: PropTypes.bool.isRequired,
  projectKey: PropTypes.string.isRequired,
  children: PropTypes.func.isRequired,
  updateQuoteState: PropTypes.func.isRequired,
  showNotification: PropTypes.func.isRequired,
  goToListRoute: PropTypes.string.isRequired,
  hasCompany: PropTypes.bool.isRequired,
  employeeEmail: PropTypes.string,
};

export default connect(null, mapDispatchToProps)(QuoteDetailsWrapper);
