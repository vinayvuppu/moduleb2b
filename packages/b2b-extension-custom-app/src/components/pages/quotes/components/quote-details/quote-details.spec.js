import React from 'react';
import { shallow } from 'enzyme';
import {
  LoadingSpinner,
  SecondaryButton,
  PrimaryButton,
} from '@commercetools-frontend/ui-kit';
import { render, fireEvent, wait } from '@testing-library/react';
import router, { MemoryRouter } from 'react-router-dom';
import { PageNotFound } from '@commercetools-frontend/application-components';
import ViewHeader from '@commercetools-local/core/components/view-header';
import TabContainer from '@commercetools-local/core/components/tab-container';

import { QUOTE_TYPES } from '../../constants';
import { QuoteDetails } from './quote-details';

jest.mock('react-router-dom', () => {
  const actual = jest.requireActual('react-router-dom');
  return {
    ...actual,
    useRouteMatch: () => ({
      url: 'www.foo.com',
      params: { projectKey: 'foo' },
    }),
    useHistory: () => ({ history: jest.fn() }),
  };
});
jest.mock('@commercetools-frontend/permissions', () => {
  const actual = jest.requireActual('@commercetools-frontend/permissions');
  return {
    ...actual,
    RestrictedByPermissions: ({ children }) => children,
  };
});
jest.mock('react-intl', () => {
  const actual = jest.requireActual('react-intl');
  return {
    ...actual,
    useIntl: jest.fn(() => ({
      formatMessage: jest.fn(msg => msg.defaultMessage),
    })),
  };
});

const createTestProps = props => ({
  quoteId: 'quote-id-1',
  quote: {
    id: 'quote-id-1',
    quoteState: 'submitted',
  },
  isLoading: false,
  projectKey: 'project-key-1',
  children: jest.fn(),
  updateQuoteState: jest.fn(),
  showNotification: jest.fn(),
  goToListRoute: 'route/quotes',
  hasCompany: true,
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;

  describe('when the quote is loading', () => {
    beforeEach(() => {
      props = createTestProps({ isLoading: true });
      wrapper = shallow(<QuoteDetails {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the Loading spinner', () => {
      expect(wrapper).toRender(LoadingSpinner);
    });
  });

  describe('when the quote is not found', () => {
    beforeEach(() => {
      props = createTestProps({ isLoading: false, quote: undefined });
      wrapper = shallow(<QuoteDetails {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the PageNotFound', () => {
      expect(wrapper).toRender(PageNotFound);
    });
  });

  describe('when the quote is loaded', () => {
    let renderCommandsWrapper;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<QuoteDetails {...props} />);
      renderCommandsWrapper = shallow(
        <div>{wrapper.find('ViewHeader').prop('commands')}</div>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the ViewHeader', () => {
      expect(wrapper).toRender(ViewHeader);
    });

    it('should render the TabContainer', () => {
      expect(wrapper).toRender(TabContainer);
    });

    it('should render the Approval button', () => {
      expect(renderCommandsWrapper).toRender(SecondaryButton);
    });

    it('should render the Decline button', () => {
      expect(renderCommandsWrapper).toRender(PrimaryButton);
    });

    describe('when the quote is not in the requested state', () => {
      beforeEach(() => {
        props = createTestProps({
          quote: {
            id: 'quote-id-1',
            quoteState: 'rejected',
          },
        });
        wrapper = shallow(<QuoteDetails {...props} />);
        renderCommandsWrapper = shallow(
          <div>{wrapper.find('ViewHeader').prop('commands')}</div>
        );
      });

      it('should not render the Approval button', () => {
        expect(renderCommandsWrapper).not.toRender(SecondaryButton);
      });

      it('should not render the Decline button', () => {
        expect(renderCommandsWrapper).not.toRender(PrimaryButton);
      });
    });
  });
});

describe('QuoteDetails component', () => {
  test('edit quote with state initial', () => {
    const pushSpy = jest.fn();
    const props = createTestProps({
      quote: { id: 'foo', quoteState: QUOTE_TYPES.INITIAL },
    });
    jest
      .spyOn(router, 'useHistory')
      .mockImplementation(() => ({ push: pushSpy }));
    const { getByText } = render(
      <MemoryRouter>
        <QuoteDetails {...props} />
      </MemoryRouter>
    );
    const editBtn = getByText('Edit');

    fireEvent.click(editBtn);

    expect(
      pushSpy
    ).toHaveBeenCalledWith(
      '/foo/b2b-extension/my-company/quotes/new/lineitems',
      { quote: { id: 'foo', quoteState: 'initial' } }
    );
  });
  test('submit quote with state initial', async () => {
    const updateQuoteStateSpy = jest.fn();
    const showNotificationSpy = jest.fn();
    const props = createTestProps({
      quote: { id: 'foo', quoteState: QUOTE_TYPES.INITIAL },
      updateQuoteState: updateQuoteStateSpy,
      showNotification: showNotificationSpy,
    });
    const { getByText } = render(
      <MemoryRouter>
        <QuoteDetails {...props} />
      </MemoryRouter>
    );
    const submitBtn = getByText('Submit');

    fireEvent.click(submitBtn);
    await wait();

    expect(updateQuoteStateSpy).toHaveBeenCalledWith(QUOTE_TYPES.SUBMITTED);
    expect(showNotificationSpy).toHaveBeenCalledWith({
      domain: 'side',
      kind: 'success',
      text: 'Quote submitted',
    });
  });
});
