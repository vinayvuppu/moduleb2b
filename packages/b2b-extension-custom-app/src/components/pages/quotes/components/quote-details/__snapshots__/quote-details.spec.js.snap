// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering when the quote is loaded should match snapshot 1`] = `
<div
  data-track-component="QuoteDetail"
>
  <View>
    <ViewHeader
      backToList={
        <RestrictedByPermissions
          permissions={
            Array [
              "ViewOrders",
            ]
          }
        >
          <BackToList
            iconType="list"
            label="To quotes list"
            to="route/quotes"
          />
        </RestrictedByPermissions>
      }
      color="neutral"
      commands={
        <React.Fragment>
          <Inline
            alignItems="flex-end"
            justifyContent="flex-end"
            scale="m"
          >
            <RestrictedByPermissions
              permissions={
                Array [
                  "ManageCustomerGroups",
                  "ManageOrders",
                ]
              }
            >
              <React.Fragment>
                <SecondaryButton
                  iconLeft={<ReviewIcon />}
                  isToggleButton={false}
                  label="Approve"
                  onClick={[Function]}
                  theme="default"
                  type="button"
                />
                <PrimaryButton
                  iconLeft={<CheckInactiveIcon />}
                  isToggleButton={false}
                  label="Decline"
                  onClick={[Function]}
                  size="big"
                  tone="urgent"
                  type="button"
                />
              </React.Fragment>
            </RestrictedByPermissions>
          </Inline>
        </React.Fragment>
      }
      title="Quote {quoteNumber}"
      withBottomLine={true}
    >
      <withRouter(TabHeader)
        key="general"
        name="general"
        to="www.foo.com/general"
      >
        General
      </withRouter(TabHeader)>
      <withRouter(TabHeader)
        key="comments"
        name="comments"
        to="www.foo.com/comments"
      >
        Comments
      </withRouter(TabHeader)>
    </ViewHeader>
    <TabContainer
      color="surface"
    />
  </View>
  <ConfirmationDialog
    isOpen={false}
    labelPrimary={
      Object {
        "defaultMessage": "Decline",
        "id": "Quote.Details.rejectButton",
      }
    }
    labelSecondary={
      Object {
        "defaultMessage": "Cancel",
        "id": "AppKit.Shared.cancel",
      }
    }
    onCancel={[Function]}
    onClose={[Function]}
    onConfirm={[Function]}
    title="Update quote state"
  >
    <TextBody>
      <FormattedMessage
        defaultMessage="Are you sure you want to change the quote state?"
        id="Quote.Details.confirmDeleteMessage"
        values={Object {}}
      />
    </TextBody>
  </ConfirmationDialog>
</div>
`;

exports[`rendering when the quote is loading should match snapshot 1`] = `
<LoadingSpinner
  scale="l"
/>
`;

exports[`rendering when the quote is not found should match snapshot 1`] = `<PageNotFound />`;
